### PR TITLE
Drop given parameter syntax

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

File: tasty/src/dotty/tools/tasty/TastyFormat.scala [hunk 1]
```diff
@@ -163,8 +163,8 @@ Standard-Section: "ASTs" TopLevelStat*
                   POLYtype       Length result_Type NamesTypes                     -- A polymorphic method type `[NamesTypes]result`, used in refinements
                   METHODtype     Length result_Type NamesTypes                     -- A method type `(NamesTypes)result`, needed for refinements
                   ERASEDMETHODtype      Length result_Type NamesTypes              -- A method type `erased (NamesTypes)result`, needed for refinements
-                  GIVENMETHODtype       Length result_Type NamesTypes              -- A method type `(given NamesTypes)result`, needed for refinements
-                  ERASEDGIVENMETHODtype Length result_Type NamesTypes              -- A method type `given(erased NamesTypes)result`, needed for refinements
+                  GIVENMETHODtype       Length result_Type NamesTypes              -- A method type `(using NamesTypes)result`, needed for refinements
```
Comment (nicolasstucki) [inline tasty/src/dotty/tools/tasty/TastyFormat.scala]: Note: At some point we will need to rename `GIVENMETHODtype` to `CONTEXTMETHODtype`

File: compiler/src/dotty/tools/dotc/parsing/Tokens.scala [hunk 2]
```diff
@@ -280,12 +280,6 @@ object Tokens extends TokensCommon {
    */
   final val startParamTokens: BitSet = modifierTokens | BitSet(VAL, VAR, AT)
 
-  /** Faced with the choice of a type `(...)` or a parameter or given type list
-   *  in `(...)`, the following tokens after the opening `(` determine it's
-   *  a parameter or given type list.
-   */
-  final val startParamOrGivenTypeTokens: BitSet = startParamTokens | BitSet(GIVEN, ERASED)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/parsing/Tokens.scala]: As far as I can see we still should have this for `ERASED` for

```scala
val a: (erased Ev) => Int = (erased ev: Ev) => Int
```

**Comment by nicolasstucki** [comment]
Just need to fix the community build. Otherwise LGTM.

**Comment by anatoliykmetyuk** [comment]
@nicolasstucki I've updated the following projects:

shapeless
intent
sourcecode
effpi
utest

effpi is the one that failed because it was unable to find the Serialized/T reference. The changes for all the projects are on the `using-syntax` branches.

To my knowledge, only xml-interpolator is left to be updated.

**Pull request merged**